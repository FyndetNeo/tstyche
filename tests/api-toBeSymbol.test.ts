import { expect, test } from "@jest/globals";
import { spawnTyche } from "./__utils__/spawnTyche.js";

test("toBeSymbol", () => {
  const { status, stderr, stdout } = spawnTyche("api-toBeSymbol");

  expect(stdout).toMatchSnapshot("stdout");
  expect(stderr).toMatchSnapshot("stderr");

  expect(status).toBe(1);
});
