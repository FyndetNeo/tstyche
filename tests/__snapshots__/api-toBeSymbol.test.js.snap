// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toBeSymbol: stderr 1`] = `
"Error: Type 'symbol' is not identical to type 'void'.

   7 |   expect(returnsSymbol()).type.toBeSymbol();
   8 | 
>  9 |   expect(returnsVoid()).type.toBeSymbol();
     |                              ^
  10 | });
  11 | 
  12 | test("is NOT symbol?", () => {

       at ./__typetests__/toBeSymbol.tst.ts:9:30 ❭ is symbol?

Error: Type 'symbol' is identical to type 'symbol'.

  13 |   expect(returnsVoid()).type.not.toBeSymbol();
  14 | 
> 15 |   expect(returnsSymbol()).type.not.toBeSymbol();
     |                                    ^
  16 | });
  17 | 

       at ./__typetests__/toBeSymbol.tst.ts:15:36 ❭ is NOT symbol?

"
`;

exports[`toBeSymbol: stdout 1`] = `
"uses TypeScript <<version>> with ./tsconfig.json

fail ./__typetests__/toBeSymbol.tst.ts
  × is symbol?
  × is NOT symbol?

Targets:    1 failed, 1 total
Test files: 1 failed, 1 total
Tests:      2 failed, 2 total
Assertions: 2 failed, 2 passed, 4 total
Duration:   <<timestamp>>

Ran all test files.
"
`;
